### PR TITLE
Completion for available section after { and {#

### LINKED ISSUE #538: Completion for available section after `{`  and {#
Today we provide snippet to generate quickly section, see https://github.com/redhat-developer/vscode-quarkus/blob/master/docs/qute/TemplateSupport.md#completion-for-qute-section

But it should be nice to provide:

 * Code completion on `{` does not propose #
 * Code completion on `{#` does not propose if

### PR DESCRIPTION
**Pull request by angelozerr**
Completion for available section after { and {#

Fixes #538

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
It should work. I need to write tests.

**Comment by fbricon** [comment]
s/suffixToFound/suffixToFind/g

**Comment by angelozerr** [comment]
The PR is now ready. Code is cleaned and tests has been written.

**Comment by angelozerr** [comment]
Here a little demo:

QuteCompletionTagDemo

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/commons/snippets/SnippetRegistry.java [hunk 1]
```diff
@@ -235,11 +235,16 @@ private static Snippet createSnippet(JsonReader reader,
 	 * @param canSupportMarkdown true if markdown is supported to generate
 	 *                           documentation and false otherwise.
 	 * @param contextFilter      the context filter.
+	 * @param suffixToFind      suffix to found to eat it when completion snippet
```
Comment (fbricon) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/commons/snippets/SnippetRegistry.java:238]: suffix to find, that will be removed when completion snippet is applied

Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/commons/snippets/SnippetRegistry.java]: fixed

**Comment by fbricon** [comment]
Thanks @angelozerr

**Pull request merged**